mlflow/gateway/providers/mosaicml.py

```diff
@@ -251,3 +259,18 @@ async def embeddings(self, payload: embeddings.RequestPayload) -> embeddings.Res
                 },
             }
         )
+
+
+def _custom_token_allowance_exceeded_handling(status_code: int, detail):
+    """
+    Custom handler for specific error messages that are incorrectly set as server-side errors, but
+    are in actuality an issue with the request sent to the external provider.
+    """
+
+    message = detail.get("message", {})
+    if (
+        status_code == 500
+        and message
+        and any(message.startswith(x) for x in MLFLOW_AI_GATEWAY_MOSAIC_ML_HANDLER_MESSAGES)
```

Comment:
I'd remove `MLFLOW_AI_GATEWAY_MOSAIC_ML_HANDLER_MESSAGES` and directly use `(
    "Error: max output tokens is limited to",
    "Error: prompt token count",
)`.